### DISCUSSION TITLE
Initial setup road bumps

### DISCUSSION THREAD

**Discussion by ChristopherHaws** [nx-dotnet/nx-dotnet#691] [General]
Hi there!

First off, thanks for creating this tool! I have been struggling to get .NET to play nicely in a JS/TS monorepo and this tool has helped quite a bit!

I wanted to document some of the hurdles I encountered when setting up my repo in case it helps others or possible gets fixed directly in nx-dotnet.

If/when I run into more hurdles, I will update this discussion with them in hopes it might help someone else! ^_^

## Make sure you create a `global.json` file prior to running `init`

I struggled for a while trying to figure out why `nx g @nx-dotnet/core:init` was failing every time I ran it. It turned out it was because I have the .NET 8 Preview SDK installed on my computer, and it was being used by default since I didn't have a `global.json` file. The fix was to run the following command prior to running the `init` command to specify the .NET SDK version to use:

```shell
dotnet new globaljson --sdk-version=7.0.200 --roll-forward=latestFeature
```

One possible solution to this would be to have `@nx-dotnet/core:init` detect if there is no global.json file and create one automatically for the user.

## `nx serve my-api` was failing to start `dotnet watch`

When I initially tried to run the app, it wouldn't start up due to duplicate global assembly attributed being present. I see that the `bin` and `obj` folders are set to use a different path than the default path, which is fine, however the templates apparently generate `bin` and `obj` folders anyway when first installed.

The solution to this was super simple, but it took me a few minutes to realize what the cause was. To fix this, just manually delete the `bin` and `obj` folders. Since `Directory.Build.props` specifies a different location for these files, they shouldn't get generated again, so the `serve` command will start working.

On a side note, use a top level folder for build artifacts is going to be getting much simpler in .NET 8. See [this issue](https://github.com/dotnet/designs/pull/281) for more info! :)

## `@nx-dotnet/core:app` vs `@nx-dotnet/core:application`
I am a bit confused about the naming here. Are these the same generators? In some places I see `app` being used and in others I see `application` being used, but I can't find any info about `app` in the docs.

## Running specific `launchSettings.json` profiles
I can't figure out how to get `nx serve` to use a specific launch profile. The defaults that get created are `http` and `https` and the one that is being used by default is `http` which causes the swagger endpoints to fail since there is no ssl endpoint. I tried running `nx serve my-api --launch-profile https` but this doesn't seem to fix the issue.

**Comment by AgentEnder** [comment]
Hey @ChristopherHaws! Thanks so much for writing this up, it helps a lot to understand what folks with a fresh perspective are seeing.

> I struggled for a while trying to figure out why nx g @nx-dotnet/core:init was failing every time I ran it.

Can you grab the error that you were encountering and open an issue for .NET 8 SDK support? It will become stable at some point, and we will need to support it at that time anyways. Currently, we don't run tests against anything after .NET 7.

> nx serve my-api was failing to start dotnet watch

This is an interesting point - I'll see if thats something we can clean up when we are scaffolding the app.

> Running specific launchSettings.json profiles

I'll look into this, I thought that the --launch-profile flag should handle it.

> @nx-dotnet/core:app vs @nx-dotnet/core:application

They are aliases of each other, but the docs should probably be consistent with `application` - you can see the alias is defined here: https://github.com/nx-dotnet/nx-dotnet/blob/master/packages/core/generators.json#L17
